### PR TITLE
PR for adding and integrating a chain switcher 

### PR DESCRIPTION
**Pull request by souvikmishra**

### CODE REVIEW

File: src/lib/components/header/sub-components/ChainSwitcher.svelte [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	import { environment } from '$lib/data-stores/environment';
+	import { walletStore } from '$lib/data-stores/walletProviderStore';
+	import { chainStore } from '$lib/data-stores/chainProviderStore';
+	import { page } from '$app/stores';
+
+	function handleChainSwitch(chainId: number) {
+		switchChain(provider, environment.valid_chains[chainId].chain_id);
+	}
+
+	$: baseRoute = $page?.route?.id?.split('/')?.[1];
```
Comment (prateekreddy) [inline src/lib/components/header/sub-components/ChainSwitcher.svelte:17]: will the base route be same as the one in config ?

File: src/lib/data-stores/toastStore.ts [hunk 2]
```diff
@@ -38,7 +38,6 @@ export const addToast = ({
 
 	const defaults: Toast = {
 		variant: 'info',
-		dismissible: true,
```
Comment (prateekreddy) [inline src/lib/data-stores/toastStore.ts:41]: Why can't it be dismissed ?

Comment (souvikmishra) [reply src/lib/data-stores/toastStore.ts:41]: here the value of `dismissible` comes from then function's arguments. This line always setting it to `true` even if the component was being passed `false` to it

File: src/lib/components/header/sub-components/ChainSwitcher.svelte (hunk 1 above)
Comment (souvikmishra) [reply src/lib/components/header/sub-components/ChainSwitcher.svelte:17]: yes, the `$page` store here just gives a ton of metadata about the current page the user is on i.e. the `+page.svelte` file that the user is currently seeing.

**Review by prateekreddy — DISMISSED** [review]

**Review by prateekreddy — APPROVED** [review]

**Pull request merged**